medusa/network/hostname_resolver.py

```diff
@@ -23,16 +26,33 @@ def __init__(self, resolve_addresses, k8s_mode):
         self.k8s_mode = k8s_mode
 
     def resolve_fqdn(self, ip_address=''):
+        logging.info(f"Resolving ip address {ip_address}")
         ip_address_to_resolve = ip_address if ip_address != '' else socket.gethostbyname(socket.getfqdn())
-
+        logging.info(f"ip address to resolve {ip_address_to_resolve}")
         if str(self.resolve_addresses) == "False":
             logging.debug("Not resolving {} as requested".format(ip_address_to_resolve))
             return ip_address_to_resolve
 
-        fqdn = socket.getfqdn(ip_address_to_resolve)
-        returned_fqdn = fqdn
-        if self.k8s_mode and fqdn.find('.') > 0:
-            returned_fqdn = fqdn.split('.')[0]
-        logging.debug("Resolved {} to {}".format(ip_address_to_resolve, returned_fqdn))
+        hostname = socket.getfqdn(ip_address_to_resolve)
+        if self.k8s_mode:
+            hostname = self.compute_k8s_hostname(ip_address_to_resolve)
+        logging.debug("Resolved {} to {}".format(ip_address_to_resolve, hostname))
+
+        return hostname
+
+    def compute_k8s_hostname(self, ip_address):
+        if (self.is_ipv4(ip_address)):
```

Comment:
What happens if the machine has IPv6/IPv4 dual stack? Can the socket.gethostbyname return IPv6 first?

Reply:
Good call, I'll add support for ipv6 as well.